variables.tf

```diff
@@ -35,3 +35,35 @@ variable "region" {
     error_message = "Invalid input, options: \"us-south\", \"ca-tor\", \"eu-es\", \"eu-de\"."
   }
 }
+
+variable "cos_instance_crn" {
+  type        = string
+  description = "CRN of the Cloud Object Storage to store SCC data"
+
+  validation {
+    condition     = var.cos_instance_crn != null
+    error_message = "Please provide COS instance CRN to store SCC data"
+  }
```

Comment:
you dont need this validation. By giving no default value, it makes the variable required.